### DISCUSSION TITLE
Convert SVG to PNG Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by Unungentium** [ImageMagick/ImageMagick#2397] [Help]
Hello!

This is my first time posting about anything to do with magick so please be understanding with your answers! 
I am trying to convert and SVG file to a PNG file. The error that pops up says
"convert: no images defined `ttbarplot.png' @ error/convert.c/ConvertImageCommand/3285."
I installed magick from a tarball if that makes any difference.

Thanks for your help!

**Comment by fmw42** [comment]
What was your exact command line?  What is your IM version and platform? Does

`convert logo: logo.gif`

work?

**Reply by Unungentium** [reply]
Hello Fred!
Thank you for your quick answer! I'm running IM version 7 I believe. Nothing actually happened when i typed in 
`convert logo: logo.gif`
What can I do from here?
Thank you!
EDIT:
My code input is 
`magick convert image.svg image.png`

**Reply by fmw42** [reply]
What is your platform?  What do you get from

`magick -version`

Try

`magick logo: logo.gif`

You should find logo.gif in your current working directory from which you issued that command.

With IM 7, you should use magick, not magick convert.  All the other tools need magick preface such as magick identify. But convert is replaced by magick.

**Reply by Unungentium** [reply]
When typing in `magick -version` I get
`Version: ImageMagick 7.0.10-26 Q16 x86_64 2020-08-10 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): zlib`
When typing `magic logo: logo.gif` I do end up getting the logo!
I then typed in `magick image.svg image2.png` and got the following error
`magick: no images for write '-write' 'image2.png' at CLI arg 2 @ error/operation.c/CLINoImageOperator/4861.`

**Reply by fmw42** [reply]
"Delegates (built-in): zlib"  Shows no delegate library for PNG.  You basically have no delegates.  How did you install Imagemagick?  If you installed from binary, then something is wrong. If you installed from source, then you must install all your desired delegates separately before installing ImageMagick.  You never answered my question about your platform!

**Reply by Unungentium** [reply]
Right sorry! I'm running it on Ubuntu (Linux). I installed it from a `.tar.gz` compressed archive. I'll have a look at how to install the delegates. Thanks for your help!

**Reply by fmw42** [reply]
So was that archive a binary or a source code? You should install from a binary if you can, because it installs the delegates for you.

See https://askubuntu.com/questions/1042436/how-to-install-delegate-libraries-for-image-magick-7-0-7
